cmd/kured/main.go

```diff
@@ -124,9 +128,49 @@ func newCommand(name string, arg ...string) *exec.Cmd {
 	return cmd
 }
 
+func getPIDtoRunCmds() int {
+	// find the pid to enter host mount space
+	statPath := fmt.Sprintf("/proc/1/stat")
+	dataBytes, err := ioutil.ReadFile(statPath)
+	if err != nil {
+		log.Infof("nil on /proc/1/stat")
+		return 1 // default to pid = 1 but will ocours error in next exec command
+	}
+	data := string(dataBytes)
+	if strings.Contains(strings.Split(data, " ")[1], "systemd") {
+		// log.Info("is systemd")
+		return 1 // is systemd
+	}
+	tstPidCmd := exec.Command("ps", "-e", "-o", "pid,comm")
+	var out bytes.Buffer
+	tstPidCmd.Stdout = &out
+	if err := tstPidCmd.Run(); err != nil {
+		log.Infof("error on run ps -e -o pid,cmd")
+		return 1 // default to 1 by will ocours error in next exec command
+	}
+	pss := strings.Split(out.String(), "\n")
+	for _, n := range pss {
+		ns := strings.Split(strings.Trim(n, " "), " ")
+		if len(ns) > 1 {
+			if strings.Contains(ns[1], "acpid") {
+				i, err := strconv.Atoi(strings.Trim(ns[0], " "))
+				if err != nil {
+					log.Infof("error on convert pid to int")
+					return 1
+				}
+				// log.Info(fmt.Sprintf("Ok pid  (%d)", ns[0]))
+				return i
+			}
+		}
+	}
+	log.Info("pid = 1")
+	return 1
+}
+
 func sentinelExists() bool {
+	pid := getPIDtoRunCmds()
```

Comment:
Instead of repeating this getPIDtoRunCmds() in sentinelExists and in commandReboot, why don't we call this in rebootAsRequired, to store what is the nsenter command to run, and pass it to the appropriate functions? (or just pass the namespace with pid, whichever one is clearer)

Reply:
The value is required in maintainRebootRequiredMetric() too, i put it in the root()

Reply:
It is ok? I sent a new commit but occurs a conflict... have a import "context" in the master not used!

Reply:
Other option is to set a global variable with the pid in the beginning of module (and keepeng the 'default' pid 1 in case of any problem because is the pid used now).